internal/controller/children.go

```diff
@@ -101,14 +102,34 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 			if !ok {
 				return fmt.Errorf("Could not cast when reconciling")
 			}
+			events := v1.EventList{}
+			selector, err := fields.ParseSelector(fmt.Sprintf("involvedObject.name=%s,involvedObject.namespace=%s,involvedObject.kind=StatefulSet", current.ObjectMeta.Name, current.ObjectMeta.Namespace))
+			if err == nil {
+				if err := clnt.List(ctx, &events, &client.ListOptions{FieldSelector: selector}); err == nil {
+					for _, event := range events.Items {
+						if event.Reason == "FailedCreate" && strings.Contains(event.Message, "exceeded quota") {
+							statusCallback = func(status *amaltheadevv1alpha1.AmaltheaSessionStatus) {
+								status.State = amaltheadevv1alpha1.Failed
+								status.Error = "Quota exceeded"
```

Comment:
suggestion: Please add a more descriptive message here that a not-so-technical user would understand. If I saw this message I would be wondering what the quota refers to (i.e. for compute resources). Also what can I do to fix this. We can provide some general guidance here. For example request less resources or contact your administrator.